L1Trigger/Phase2L1ParticleFlow/interface/DiscretePFInputs.h

```diff
@@ -0,0 +1,244 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_DiscretePFInputs_H
+#define L1Trigger_Phase2L1ParticleFlow_DiscretePFInputs_H
+
+#if defined(__GXX_EXPERIMENTAL_CXX0X__) or defined(CMSSW)
+#include <cstdint>
+#include <limits>
+#define L1Trigger_Phase2L1ParticleFlow_DiscretePFInputs_MORE
+#else
+#include <stdint.h>
+#endif
+
+namespace l1t {
+  class PFTrack;
+  class PFCluster;
+  class PFCandidate;
+  class Muon;
+}  // namespace l1t
+
+// the serialization may be hidden if needed
+#include <cmath>
+#include <vector>
+
+namespace l1tpf_impl {
+
+  struct CaloCluster {
+    int16_t hwPt;
+    int16_t hwEmPt;
+    int16_t hwPtErr;
+    int16_t hwEta;
+    int16_t hwPhi;
+    uint16_t hwFlags;
+    bool isEM, used;
+    const l1t::PFCluster *src;
+
+    // sorting
+    bool operator<(const CaloCluster &other) const { return hwPt > other.hwPt; }
+
+#ifdef L1Trigger_Phase2L1ParticleFlow_DiscretePFInputs_MORE
```

Comment:
I have to recheck with newer versions of Vivado, but older ones had at times issues with c++0x

Reply:
if the issue is with the level of c++ available in the outside compilation, it could make sense to check something like
```
#if __cplusplus >= 201103L
```